airbyte-integrations/connectors/source-iterable/source_iterable/api.py

```diff
@@ -0,0 +1,231 @@
+
+
+class Lists(IterableStream):
+    data_field = "lists"
+
+    def path(self, **kwargs) -> str:
+        return "lists"
+
+
+class ListUsers(IterableStream):
+    data_field = "getUsers"
+    name = "list_users"
+
+    def path(self, parent_stream_record, **kwargs) -> str:
+        return f"lists/{self.data_field}?listId={parent_stream_record['id']}"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        list_id = self._get_list_id(response.url)
+        for user in response.iter_lines():
+            yield {"email": user.decode(), "listId": list_id}
+
+    @staticmethod
+    def _get_list_id(url: str) -> int:
+        parsed_url = urlparse.urlparse(url)
+        for q in parsed_url.query.split("&"):
+            key, value = q.split("=")
+            if key == "listId":
+                return int(value)
+
+
+class Campaigns(IterableStream):
+    data_field = "campaigns"
+
+    def path(self, **kwargs) -> str:
+        return "campaigns"
+
+
+class Channels(IterableStream):
+    data_field = "channels"
+
+    def path(self, **kwargs) -> str:
+        return "channels"
+
+
+class EmailBounce(IterableExportStream):
+    name = "email_bounce"
+    data_field = "emailBounce"
+
+
+class EmailClick(IterableExportStream):
+    name = "email_click"
+    data_field = "emailClick"
+
+
+class EmailComplaint(IterableExportStream):
+    name = "email_complaint"
+    data_field = "emailComplaint"
+
+
+class EmailOpen(IterableExportStream):
+    name = "email_open"
+    data_field = "emailOpen"
+
+
+class EmailSend(IterableExportStream):
+    name = "email_send"
+    data_field = "emailSend"
+
+
+class EmailSendSkip(IterableExportStream):
+    name = "email_send_skip"
+    data_field = "emailSendSkip"
+
+
+class EmailSubscribe(IterableExportStream):
+    name = "email_subscribe"
+    data_field = "emailSubscribe"
+
+
+class EmailUnsubscribe(IterableExportStream):
+    name = "email_unsubscribe"
+    data_field = "emailUnsubscribe"
+
+
+class MessageTypes(IterableStream):
+    data_field = "messageTypes"
+    name = "message_types"
+
+    def path(self, **kwargs) -> str:
+        return "messageTypes"
+
+
+class Metadata(IterableStream):
+    data_field = "results"
+
+    def path(self, **kwargs) -> str:
+        return "metadata"
+
+
+class Templates(IterableExportStream):
+    data_field = "templates"
+    template_types = ["Base", "Blast", "Triggered", "Workflow"]
+    message_types = ["Email", "Push", "InApp", "SMS"]
+
+    def path(self, **kwargs) -> str:
+        return "templates?templateType={}&messageMedium={}"
+
+    def _list_records(self, stream_state: Mapping[str, Any], parent_stream_record: Mapping[str, Any] = None) -> Iterable[Mapping[str, Any]]:
+        """
+        :param parent_stream_record If this is a child stream, this is a record from the parent stream. Otherwise, this record is None.
+        :return:
+        """
+
+        for template in self.template_types:
```

Comment:
hmmm I see why you had to do this...not ideal that we are overriding an `_` method. Working on an upcoming refactor that will hopefully make this much cleaner.

Reply:
I think I solved this problem and used only the `read_stream` method with setting `stream_params`, please look at the PR after the update.